ResultadosDigitais/matrix
# Adds internationalization

### Description
Well guys get ready for this one, it's a pretty big PR.

![matrix](https://user-images.githubusercontent.com/12252332/87861795-f8967300-c91f-11ea-9879-8d5d42049c19.gif)
(Ignore the color twitching, I had to compress the gif a lot.)

---

This PR adds internationalization feature for the Matrix.

I extracted all component strings and put them in language files under the new `src/locales` folder, I also translated all of them to portuguese.

This is the proposed file structure, which is the commonly used across applications with i18n.

![image](https://user-images.githubusercontent.com/12252332/87861816-50cd7500-c920-11ea-948f-2b395cacf9f0.png)

There are some other strings that I couldn't translate, which are not within components, so that is something we'll have to work on until this PR can be merged.

I had to do some refactoring in some parts of the code so I could use i18n, and I also had to replace the Google SignIn button so I could translate the text. I also changed some other things that I deemed necessary (some a11y stuff, for example).

Some tests are currently failing and I will work on that as soon as I can find a way to mock i18next or just use it in the tests.

I also didn't add anything to the backend, so if there are any messages there that need translation, that's also something that needs to be worked on.

### How to test?
Once you open the matrix, the application will now try to check what's the preferred language using `navigator.language` and fallback to the default locale (`en`) if the language is not supported by matrix.  

When logged in, there's a new button in the appbar that opens the language switcher menu, which currently has both portuguese and english as options. Clicking them will change the application language and store it in the `localStorage`, so the language choice persists when the user refreshes the page.
